docs/user/ai_model_serving.md

```diff
@@ -0,0 +1,482 @@
+
+```sh
+$ oc expose svc -n ai-demo ovms-resnet50-predictor
+route.route.openshift.io/ovms-resnet50-predictor exposed
+
+$ oc get route -n ai-demo
+NAME                      HOST                                               ADMITTED   SERVICE                   TLS
+ovms-resnet50-predictor   ovms-resnet50-predictor-ai-demo.apps.example.com   True       ovms-resnet50-predictor
+```
+
+### Querying the model server
+
+Finally, we'll check if our model is ready for inference and we'll reuse OVMS examples to test the inference.
+
+Get the IP of the MicroShift cluster and assign it to the `IP` variable.
+Use the HOST value of the Route created in previous step and assign to the `DOMAIN` variable.
+Next, run the following `curl` command.
+Alternatively, instead of using `--connect-to "${DOMAIN}::${IP}:"`,
+you can use real DNS, or add the IP and the domain to the `/etc/hosts` file.
+```sh
+DOMAIN=ovms-resnet50-predictor-ai-demo.apps.example.com
+IP=192.168.0.10
+curl -i "${DOMAIN}/v2/models/ovms-resnet50/ready" \
+    --connect-to "${DOMAIN}::${IP}:"
+```
+
+Response code 200 is expected. Example output:
+```
+HTTP/1.1 200 OK
+content-type: application/json
+date: Wed, 12 Mar 2025 16:01:32 GMT
+content-length: 0
+set-cookie: 56bb4b6df4f80f0b59f56aa0a5a91c1a=4af1408b4a1c40925456f73033d4a7d1; path=/; HttpOnly
+```
+
+We can also query model's metadata:
+```sh
+curl "${DOMAIN}/v2/models/ovms-resnet50" \
+    --connect-to "${DOMAIN}::${IP}:"
+```
+Output:
+```json
+{"name":"ovms-resnet50","versions":["1"],"platform":"OpenVINO","inputs":[{"name":"0","datatype":"FP32","shape":[1,224,224,3]}],"outputs":[{"name":"1463","datatype":"FP32","shape":[1,1000]}]
+```
+
+Let's try querying actual model - following example verifies if inference is in accordance to the training data.
+
+First, download an image of a bee from the OpenVino examples:
+```sh
+curl -O https://raw.githubusercontent.com/openvinotoolkit/model_server/main/demos/common/static/images/bee.jpeg
+```
+
+Next, we need to create the request data.
+1. Start with an inference header in JSON format.
+1. Get the size of the header. It needs to be passed to the OVMS later in form of HTTP header.
+1. Append size of the image to the request file. OVMS expects 4 bytes (little endian).
+   Command below uses `xxd` utility which is part of `vim-common` package.
+1. Append the image to the request file.
+
+```sh
+IMAGE=./bee.jpeg
+REQ=./request.json
+
+# Add an inference header
+echo -n '{"inputs" : [{"name": "0", "shape": [1], "datatype": "BYTES"}]}' > "${REQ}"
+
+# Get the size of the inference header
+HEADER_LEN="$(stat -c %s "${REQ}")"
+
+# Add size of the data (image) in binary format (4 bytes, little endian)
+printf "%08X" $(stat --format=%s "${IMAGE}") | sed 's/\(..\)/\1\n/g' | tac | tr -d '\n' | xxd -r -p >> "${REQ}"
+
+# Add the data, i.e. the image
+cat "${IMAGE}" >> "${REQ}"
+```
+
+Now we can make an inference request against the model server using `ovms-resnet50` model.
```

Comment:
```suggestion
Now we can make an inference request against the model server that is using the `ovms-resnet50` model.
```
it's the server using the model; not us using the model to make the inference request, right?

Reply:
Yes, we create an input data and send it to the model server, model server takes a request and uses model to process the requests.